src/test/java/com/wikia/webdriver/testcases/adstests/TestUrlBuilder.java

```diff
@@ -1,378 +1,420 @@
-            Assert.assertEquals(c.newInstance((String) data[0], "preview", (Boolean) data[2], (Boolean) data[3], DEFAULT_LANGUAGE)
-                            .getUrlForPath((String) data[1]),
-                    (String) data[5]);
-            Assert.assertEquals(c.newInstance((String) data[0], "sandbox", (Boolean) data[2], (Boolean) data[3], DEFAULT_LANGUAGE)
-                            .getUrlForPath((String) data[1]),
-                    (String) data[6]);
-            Assert.assertEquals(c.newInstance((String) data[0], "sandbox-mercurydev", (Boolean) data[2], (Boolean) data[3], DEFAULT_LANGUAGE)
-                            .getUrlForPath((String) data[1]),
-                    (String) data[7]);
-            Assert.assertEquals(c.newInstance((String) data[0], "dev-dmytror", (Boolean) data[2], (Boolean) data[3], DEFAULT_LANGUAGE)
-                            .getUrlForPath((String) data[1]),
-                    (String) data[8]);
-            Assert.assertEquals(c.newInstance((String) data[0], "staging", (Boolean) data[2], (Boolean) data[3], DEFAULT_LANGUAGE)
-                            .getUrlForPath((String) data[1]),
-                    (String) data[9]);
-        }
+    for (Object[] data : TEST_DATA) {
+      Assert.assertEquals(
+          c.newInstance((String) data[0], "prod", (Boolean) data[2], (Boolean) data[3],
+              DEFAULT_LANGUAGE)
+              .getUrlForPath((String) data[1]),
+          (String) data[4]);
+      Assert.assertEquals(
+          c.newInstance((String) data[0], "preview", (Boolean) data[2], (Boolean) data[3],
+              DEFAULT_LANGUAGE)
+              .getUrlForPath((String) data[1]),
+          (String) data[5]);
+      Assert.assertEquals(
+          c.newInstance((String) data[0], "sandbox", (Boolean) data[2], (Boolean) data[3],
+              DEFAULT_LANGUAGE)
+              .getUrlForPath((String) data[1]),
+          (String) data[6]);
+      Assert.assertEquals(c.newInstance((String) data[0], "sandbox-mercurydev", (Boolean) data[2],
+          (Boolean) data[3], DEFAULT_LANGUAGE)
+              .getUrlForPath((String) data[1]),
+          (String) data[7]);
+      Assert.assertEquals(
+          c.newInstance((String) data[0], "dev-dmytror", (Boolean) data[2], (Boolean) data[3],
+              DEFAULT_LANGUAGE)
+              .getUrlForPath((String) data[1]),
+          (String) data[8]);
+      Assert.assertEquals(
+          c.newInstance((String) data[0], "staging", (Boolean) data[2], (Boolean) data[3],
+              DEFAULT_LANGUAGE)
+              .getUrlForPath((String) data[1]),
+          (String) data[9]);
     }
+  }
 
-    @Test(groups = {"TestUrlBuilder", "unitTests"})
-    public void urlWithLanguageBuilderTests() throws Exception {
-        Configuration.setTestValue("qs", "");
-        Constructor<UrlBuilder> c = getUrlBuilderConstructor();
+  @Test(groups = {"TestUrlBuilder", "unitTests"})
+  public void urlWithLanguageBuilderTests() throws Exception {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS00112)